### DISCUSSION TITLE
RBAC isn't being applied as expected to GitHub SSO Team

### DISCUSSION THREAD

**Discussion by mimartin12** [argoproj/argo-cd#11742] [Q&A]
I have a team in Github and ArgoCD is configured to use SSO for the GitHub org.
Inside of my argocd configmap, I have the following.

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-cm
  labels:
    app.kubernetes.io/name: argocd-cm
    app.kubernetes.io/part-of: argocd
data:
  policy.default: role:readonly
  policy.csv: |
    p, role:account-admin, accounts, get, *, allow
    p, role:account-admin, projects, get, *, allow
    p, role:account-admin, projects, create, *, allow
    p, role:account-admin, projects, update, *, allow
    p, role:account-admin, projects, delete, *, allow
    p, role:account-admin, repositories, get, *, allow
    p, role:account-admin, repositories, create, *, allow
    p, role:account-admin, repositories, update, *, allow
    p, role:account-admin, repositories, delete, *, allow
    g, github-org:argocd-admin, role:account-admin
    g, github-org:argocd-admin, role:admin # <- Default ArgoCD built-in role.
```

Based on my understanding of ArgoCD RBAC, this should do two things
1. Create a role called `account-admin` that has the ability to list accounts, manage projects, manage repositories.
2. Assigns the `account-admin` and the built-in ArgoCD role `admin` to the <github-org-name>:argocd-admin team.

However, after applying this configuration, I am unable to do the following as a member of the `argocd-admin` team.
- Create a project
- Delete a project
![image](https://user-images.githubusercontent.com/77340197/208196696-c45c4c88-3e09-42a7-af63-73fc7e198016.png)

A few other things that I can't do even though I am being assigned `role:admin`.
- List certificates
- List GnuPG public keys

I am able to
- Create a repository in the default project

Looking at ArgoCD's user info in the web dashboard, I can see that my group is accurate.
![Screenshot 2022-12-16 at 3 05 53 PM](https://user-images.githubusercontent.com/77340197/208196833-a5f68d1d-68b5-47cf-bc5c-535504b6c1fd.png)

Is there something missing here? How can I go about diagnosing this.

**Comment by crenshaw-dev** [comment]
What Argo CD version?

**Reply by mimartin12** [reply]
`Version: v2.5.5+fc3eaec`

As a follow-up, I used the ArgoCLI to validate the RBAC and this was the output.
```
argocd admin settings rbac can <github-org>:argocd-admin get certificates --policy-file components/users/argocd-rbac-policy.yaml
Yes
```

**Reply by mimartin12** [reply]
> What Argo CD version?

I also tried with version `v2.4.15+05acf7a` which is what I have been running in production for some time now. It experiences the same thing as above.

**Reply by mimartin12** [reply]
I attempted to create a local account, `deopvs` and assign the `account-admin` role to the local account. 

Account enabled
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-cm
  labels:
    app.kubernetes.io/name: argocd-cm
    app.kubernetes.io/part-of: argocd
data:
  accounts.devops: apiKey, login
  accounts.devops.enabled: "true"
```

Role Assigned
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-cm
  labels:
    app.kubernetes.io/name: argocd-cm
    app.kubernetes.io/part-of: argocd
data:
  policy.default: role:readonly
  policy.csv: |
    p, role:account-admin, accounts, get, *, allow
    p, role:account-admin, projects, get, *, allow
    p, role:account-admin, projects, create, *, allow
    p, role:account-admin, projects, update, *, allow
    p, role:account-admin, projects, delete, *, allow
    p, role:account-admin, repositories, get, *, allow
    p, role:account-admin, repositories, create, *, allow
    p, role:account-admin, repositories, update, *, allow
    p, role:account-admin, repositories, delete, *, allow
    g, devops, role:account-admin
```

I reset the password using the ArgoCLI, then attempted to create a project.
<img width="832" alt="image" src="https://user-images.githubusercontent.com/77340197/208466493-4e778061-0d36-4dac-b02d-d8fd3b48977f.png">

**Reply by Bayene** [reply]
I have exactly the same problem for three days I can't solve it I don't know how you finally solved it?

**Reply by mimartin12** [reply]
> I have exactly the same problem for three days I can't solve it I don't know how you finally solved it?

Hey @Bayene, 
This error came from me using the incorrect name for the config map. You should be using `name: argocd-rbac-cm`, making sure you set the labels correctly as well.

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-rbac-cm
  labels:
    app.kubernetes.io/name: argocd-rbac-cm
    app.kubernetes.io/part-of: argocd
data:
```

Hope this helps!
